src/openfl/text/TextField.hx

```diff
@@ -2934,59 +2944,6 @@ class TextField extends InteractiveObject implements IShaderDrawable {
 				__stopCursorTimer ();
 				__startCursorTimer ();
 			
-			case C:
-				
-				if (#if mac modifier.metaKey #elseif js modifier.metaKey || modifier.ctrlKey #else modifier.ctrlKey #end) {
-					
-					if (__caretIndex != __selectionIndex) {
-						
-						Clipboard.text = __text.substring (__caretIndex, __selectionIndex);
-						
-					}
-					
-				}
-			
-			case X:
-				
-				if (#if mac modifier.metaKey #elseif js modifier.metaKey || modifier.ctrlKey #else modifier.ctrlKey #end) {
-					
-					if (__caretIndex != __selectionIndex) {
-						
-						Clipboard.text = __text.substring (__caretIndex, __selectionIndex);
-						
-						replaceSelectedText ("");
-						dispatchEvent (new Event (Event.CHANGE, true));
-						
-					}
-					
-				}
-			
-			#if !js
-			case V:
-				
-				if (#if mac modifier.metaKey #else modifier.ctrlKey #end) {
-					
-					var text = Clipboard.text;
-					
-					if (text != null) {
-						
-						replaceSelectedText (text);
-						
-					} else {
-						
-						replaceSelectedText ("");
-						
-					}
-					
-					dispatchEvent (new Event (Event.CHANGE, true));
-					
-				} else {
-					
-					__textEngine.textFormatRanges[__textEngine.textFormatRanges.length - 1].end = __text.length;
-					
-				}
-			#end
```

Comment:
Isn't this still needed for non js code? So why not just extend the js fencing of V to X and C?

Reply:
I'm pretty sure everything non-js is already broken anyway, but if we'd want this to work for non-js we should implement `onTextCut/Copy/Paste` dispatch on the Lime level just like we did for HTML5. OpenFL is supposed to be higher level than this.

Reply:
I disagree, on targets where there are no cut/copy/paste events triggered by the system, it still makes sense to simulate this behavior here. We do this also for other shortcuts (e.g. Ctrl+A) because there is no such system event (at least not for HTML5). So I would leave it for targets that have to simulate missing events

Reply:
I guess it's another ticket, but you still have the non-editable but selectable branch with keydown CTRL+C handling and lime clipboard workaround here: https://github.com/innogames/openfl/pull/114/files#diff-a6e42bb0f702e824dad4f661af6ac18eR2704

I'm totally fine with implementing this in another ticket, but I'm not sure if it's actually broken with the other changes on lime side now?

Reply:
Well, it's actually a bit of a difficult discussion. Because keeping this to me make sense, but it could also be done by additional keyboard handling on lime side to send the new (and nicer) events platforms missing these system events.

This might actually also make sense for CTRL+A and HTML5, to create another event and send it from lime by making the keyboard handling on HTML5Window. But well, making everything as best as possible for all targets is not the right time.

Reply:
> on targets where there are no cut/copy/paste events triggered by the system, it still makes sense to simulate this behavior here

I think the behaviour should be simulated and transformed into the events on the Lime level otherwise there's no point in Lime at all, if OpenFL has to deal with all this via the `#if`-fest.

I mean, there's a special "text events" API in Lime that handles input (and now copy/pasting) which makes sense to me and I don't see why OpenFL should be aware of platforms here instead of the implementation of that API.

Reply:
> I guess it's another ticket, but you still have the non-editable but selectable branch with keydown CTRL+C handling and lime clipboard workaround here:

Yes, I mentioned it here: https://github.com/innogames/openfl/pull/114#issuecomment-517222010

It's not broken with the Lime changes because we don't enable text events for non-editable TextFields right now. We'll have to rework this once again to enable some of the events.

Reply:
> I mean, there's a special "text events" API in Lime that handles input (and now copy/pasting) which makes sense to me and I don't see why OpenFL should be aware of platforms here instead of the implementation of that API.

Yes, that's the thing. The more I think about it it's a bit difficult.

It's the lime window/application which forwards browser/system events from lime to openfl. And sometimes some events are existing and sometimes not. And if not they either cannot be send to can be simulated by other events (like we do for CTRL+A for select all yb keyboard events). But maybe it would be right to have "select all" as part of this text event API. Same for "delete", or "undo/redo" as you mentioned. Not sure if even all the navigation events would ideally move to lime side to be transferred into individual events.

Anyhow, I see your point to remove this block for now. And we have to revisit #114

Reply:
Moving CTRL+A and friends to lower level _might_ make sense, but for now I'm fine with how it is because at least it's consistently handled in one place. What you're proposing is handling clipboard stuff differently on two separate levels depending on the platform and that I don't like.

Reply:
It makes more sense to stop this discussion for now and to continue in person once I'm back in the office tomorrow. My original point was, you changed the API to avoid clipboard API usage for cut/copy. This can be a valid approach, but it also means we need to implement cut/copy differently for other targets now.

And this goes down to the disusssion again that I don't like to break other targets if not really necessary. And I think the approach I proposed on lime side (still use clipboard API instead of data provider callback) could help to avoid breaking other targets for now. It's not about that I don't like the data provider approach and it's not that I don't see many valid points of further extending/improving text input API in general. I just prefer if changes are less intrusive - especially if the alternative is not significantly worst.

As said, let's discuss it tomorrow.